packages/next/src/lib/create-router-client-filter.ts

```diff
@@ -0,0 +1,71 @@
+import { BloomFilter } from '../shared/lib/bloom-filter'
+import { isDynamicRoute } from '../shared/lib/router/utils'
+import { removeTrailingSlash } from '../shared/lib/router/utils/remove-trailing-slash'
+import { Redirect } from './load-custom-routes'
+
+const POTENTIAL_ERROR_RATE = 0.02
+
+export function createClientRouterFilter(
+  paths: string[],
+  redirects: Redirect[]
+): {
+  staticFilter: ReturnType<BloomFilter['export']>
+  dynamicFilter: ReturnType<BloomFilter['export']>
+} {
+  const staticPaths = new Set<string>()
+  const dynamicPaths = new Set<string>()
+
+  for (const path of paths) {
+    if (isDynamicRoute(path)) {
+      let subPath = ''
+      const pathParts = path.split('/')
+
+      for (let i = 1; i < pathParts.length + 1; i++) {
```

Comment:
Why does this loop start from 1?

Reply:
The path starts with `/` so first item is an empty string when split

Reply:
Maybe add a comment for that 👍

Reply:
comment added in https://github.com/vercel/next.js/pull/46290/files#diff-df2486624ca574a2f925d5754ef88cffffdaff6b76d888eb99e4185d862f9a11R23